ticketbuyer/buyer.go

```diff
@@ -454,34 +464,49 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
 	// For spreading your ticket purchases evenly throughout window.
 	// Use available funds to calculate how many tickets to buy, and also
 	// approximate the income you're going to have from older tickets that
-	// you've voted and are maturing during this window (tixWillRedeem)
+	// you've voted and are maturing during this window
 	if t.cfg.SpreadTicketPurchases && toBuyForBlock > 0 {
 		log.Debugf("Spreading purchases throughout window")
 
+		// same as proportionlive that getstakeinfo rpc shows
+		proportionLive := float64(t.stakeLive) / float64(t.stakePoolSize)
 		// Number of blocks remaining to purchase tickets in this window
 		blocksRemaining := int(winSize) - t.idxDiffPeriod
 		// Estimated number of tickets you will vote on and redeem this window
-		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * t.ProportionLive
+		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * proportionLive
+		// Average price of your tickets in the pool
+		yourAvgTixPrice := bal.LockedByTickets.ToCoin() / float64(t.stakeLive+t.stakeImmature)
```

Comment:
So the below comment about the Nan's appear to be related to this.  Need a check if t.stakeLive && t.stakeImmature == 0